### ISSUE TITLE
Types errors

### ISSUE THREAD

**Issue by omirobarcelo** [fakoua/soxa#1]
When trying to implement the example, the following type errors appear:
```
error: TS7006 [ERROR]: Parameter 'config' implicitly has an 'any' type.
const fetchAdapter = function (config): Promise<any> {
                               ~~~~~~
    at https://deno.land/x/soxa/src/adapters/fetchAdapter.ts:5:32

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.
            h[key] = value
            ~~~~~~
    at https://deno.land/x/soxa/src/adapters/fetchAdapter.ts:49:13

TS7006 [ERROR]: Parameter 'value' implicitly has an 'any' type.
const isCancel = function (value): boolean {
                           ~~~~~
    at https://deno.land/x/soxa/src/cancel/isCancel.ts:1:28

TS7006 [ERROR]: Parameter 'fulfilled' implicitly has an 'any' type.
    use(fulfilled, rejected): number {
        ~~~~~~~~~
    at https://deno.land/x/soxa/src/core/InterceptorManager.ts:7:9

TS7006 [ERROR]: Parameter 'rejected' implicitly has an 'any' type.
    use(fulfilled, rejected): number {
                   ~~~~~~~~
    at https://deno.land/x/soxa/src/core/InterceptorManager.ts:7:20

TS2322 [ERROR]: Type 'null' is not assignable to type 'object'.
            this.handlers[id] = null;
            ~~~~~~~~~~~~~~~~~
    at https://deno.land/x/soxa/src/core/InterceptorManager.ts:16:13

TS7006 [ERROR]: Parameter 'fn' implicitly has an 'any' type.
    forEach(fn): any {
            ~~
    at https://deno.land/x/soxa/src/core/InterceptorManager.ts:19:13

TS7006 [ERROR]: Parameter 'h' implicitly has an 'any' type.
        utils.forEach(this.handlers, function forEachHandler(h) {
                                                             ^
    at https://deno.land/x/soxa/src/core/InterceptorManager.ts:20:62

TS7006 [ERROR]: Parameter 'instanceConfig' implicitly has an 'any' type.
    constructor(instanceConfig) {
                ~~~~~~~~~~~~~~
    at https://deno.land/x/soxa/src/core/Soxa.ts:16:17

TS7006 [ERROR]: Parameter 'config' implicitly has an 'any' type.
    request(config): Promise<any> {
            ~~~~~~
    at https://deno.land/x/soxa/src/core/Soxa.ts:29:13

TS7006 [ERROR]: Parameter 'interceptor' implicitly has an 'any' type.
        this.interceptors.request.forEach(function unshiftRequestInterceptors(interceptor) {
                                                                              ~~~~~~~~~~~
    at https://deno.land/x/soxa/src/core/Soxa.ts:51:79

TS7006 [ERROR]: Parameter 'interceptor' implicitly has an 'any' type.
        this.interceptors.response.forEach(function pushResponseInterceptors(interceptor) {
                                                                             ~~~~~~~~~~~
    at https://deno.land/x/soxa/src/core/Soxa.ts:55:78

TS7006 [ERROR]: Parameter 'config' implicitly has an 'any' type.
    getUri(config): string {
           ~~~~~~
    at https://deno.land/x/soxa/src/core/Soxa.ts:66:12

TS7006 [ERROR]: Parameter 'config' implicitly has an 'any' type.
    delete(url: string, config?): Promise<any> {
                        ~~~~~~~
    at https://deno.land/x/soxa/src/core/Soxa.ts:86:25

TS7006 [ERROR]: Parameter 'config' implicitly has an 'any' type.
    get(url: string, config?): Promise<any> {
                     ~~~~~~~
    at https://deno.land/x/soxa/src/core/Soxa.ts:90:22

TS7006 [ERROR]: Parameter 'config' implicitly has an 'any' type.
    head(url: string, config?): Promise<any> {
                      ~~~~~~~
    at https://deno.land/x/soxa/src/core/Soxa.ts:94:23

TS7006 [ERROR]: Parameter 'config' implicitly has an 'any' type.
    options(url: string, config?): Promise<any> {
                         ~~~~~~~
    at https://deno.land/x/soxa/src/core/Soxa.ts:98:26

TS7006 [ERROR]: Parameter 'config' implicitly has an 'any' type.
function throwIfCancellationRequested(config): void {
                                      ~~~~~~
    at https://deno.land/x/soxa/src/core/dispatchRequest.ts:11:39

TS7006 [ERROR]: Parameter 'config' implicitly has an 'any' type.
const dispatchRequest = function (config): Promise<any> {
                                  ~~~~~~
    at https://deno.land/x/soxa/src/core/dispatchRequest.ts:23:35

TS7006 [ERROR]: Parameter 'response' implicitly has an 'any' type.
  return adapter(config).then(function onAdapterResolution(response) {
                                                           ~~~~~~~~
    at https://deno.land/x/soxa/src/core/dispatchRequest.ts:52:60

TS7006 [ERROR]: Parameter 'reason' implicitly has an 'any' type.
  }, function onAdapterRejection(reason) {
                                 ~~~~~~
    at https://deno.land/x/soxa/src/core/dispatchRequest.ts:63:34

TS7006 [ERROR]: Parameter 'error' implicitly has an 'any' type.
const enhanceError = function (error, config, code, request, response) {
                               ~~~~~
    at https://deno.land/x/soxa/src/core/enhanceError.ts:11:32

TS7006 [ERROR]: Parameter 'config' implicitly has an 'any' type.
const enhanceError = function (error, config, code, request, response) {
                                      ~~~~~~
    at https://deno.land/x/soxa/src/core/enhanceError.ts:11:39

TS7006 [ERROR]: Parameter 'code' implicitly has an 'any' type.
const enhanceError = function (error, config, code, request, response) {
                                              ~~~~
    at https://deno.land/x/soxa/src/core/enhanceError.ts:11:47

TS7006 [ERROR]: Parameter 'request' implicitly has an 'any' type.
const enhanceError = function (error, config, code, request, response) {
                                                    ~~~~~~~
    at https://deno.land/x/soxa/src/core/enhanceError.ts:11:53

TS7006 [ERROR]: Parameter 'response' implicitly has an 'any' type.
const enhanceError = function (error, config, code, request, response) {
                                                             ~~~~~~~~
    at https://deno.land/x/soxa/src/core/enhanceError.ts:11:62

TS7006 [ERROR]: Parameter 'config1' implicitly has an 'any' type.
const mergeConfig = function (config1, config2) {
                              ~~~~~~~
    at https://deno.land/x/soxa/src/core/mergeConfig.ts:11:31

TS7006 [ERROR]: Parameter 'config2' implicitly has an 'any' type.
const mergeConfig = function (config1, config2) {
                                       ~~~~~~~
    at https://deno.land/x/soxa/src/core/mergeConfig.ts:11:40

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.
      config[prop] = config2[prop];
      ~~~~~~~~~~~~
    at https://deno.land/x/soxa/src/core/mergeConfig.ts:28:7

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.
      config[prop] = utils.deepMerge(config1[prop], config2[prop]);
      ~~~~~~~~~~~~
    at https://deno.land/x/soxa/src/core/mergeConfig.ts:34:7

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.
      config[prop] = config2[prop];
      ~~~~~~~~~~~~
    at https://deno.land/x/soxa/src/core/mergeConfig.ts:36:7

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.
      config[prop] = utils.deepMerge(config1[prop]);
      ~~~~~~~~~~~~
    at https://deno.land/x/soxa/src/core/mergeConfig.ts:38:7

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.
      config[prop] = config1[prop];
      ~~~~~~~~~~~~
    at https://deno.land/x/soxa/src/core/mergeConfig.ts:40:7

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.
      config[prop] = config2[prop];
      ~~~~~~~~~~~~
    at https://deno.land/x/soxa/src/core/mergeConfig.ts:46:7

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.
      config[prop] = config1[prop];
      ~~~~~~~~~~~~
    at https://deno.land/x/soxa/src/core/mergeConfig.ts:48:7

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.
      config[prop] = config2[prop];
      ~~~~~~~~~~~~
    at https://deno.land/x/soxa/src/core/mergeConfig.ts:64:7

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.
      config[prop] = config1[prop];
      ~~~~~~~~~~~~
    at https://deno.land/x/soxa/src/core/mergeConfig.ts:66:7

TS7006 [ERROR]: Parameter 'resolve' implicitly has an 'any' type.
const settle = function (resolve, reject, response) {
                         ~~~~~~~
    at https://deno.land/x/soxa/src/core/settle.ts:10:26

TS7006 [ERROR]: Parameter 'reject' implicitly has an 'any' type.
const settle = function (resolve, reject, response) {
                                  ~~~~~~
    at https://deno.land/x/soxa/src/core/settle.ts:10:35

TS7006 [ERROR]: Parameter 'response' implicitly has an 'any' type.
const settle = function (resolve, reject, response) {
                                          ~~~~~~~~
    at https://deno.land/x/soxa/src/core/settle.ts:10:43

TS7006 [ERROR]: Parameter 'data' implicitly has an 'any' type.
const transformData = function (data, headers, fns) {
                                ~~~~
    at https://deno.land/x/soxa/src/core/transformData.ts:11:33

TS7006 [ERROR]: Parameter 'headers' implicitly has an 'any' type.
const transformData = function (data, headers, fns) {
                                      ~~~~~~~
    at https://deno.land/x/soxa/src/core/transformData.ts:11:39

TS7006 [ERROR]: Parameter 'fns' implicitly has an 'any' type.
const transformData = function (data, headers, fns) {
                                               ~~~
    at https://deno.land/x/soxa/src/core/transformData.ts:11:48

TS7006 [ERROR]: Parameter 'fn' implicitly has an 'any' type.
  utils.forEach(fns, function transform(fn) {
                                        ~~
    at https://deno.land/x/soxa/src/core/transformData.ts:13:41

TS7006 [ERROR]: Parameter 'headers' implicitly has an 'any' type.
function setContentTypeIfUnset(headers, value) {
                               ~~~~~~~
    at https://deno.land/x/soxa/src/defaults.ts:9:32

TS7006 [ERROR]: Parameter 'value' implicitly has an 'any' type.
function setContentTypeIfUnset(headers, value) {
                                        ~~~~~
    at https://deno.land/x/soxa/src/defaults.ts:9:41

TS7006 [ERROR]: Parameter 'data' implicitly has an 'any' type.
  transformRequest: [function transformRequest(data, headers) {
                                               ~~~~
    at https://deno.land/x/soxa/src/defaults.ts:33:48

TS7006 [ERROR]: Parameter 'headers' implicitly has an 'any' type.
  transformRequest: [function transformRequest(data, headers) {
                                                     ~~~~~~~
    at https://deno.land/x/soxa/src/defaults.ts:33:54

TS7006 [ERROR]: Parameter 'data' implicitly has an 'any' type.
  transformResponse: [function transformResponse(data) {
                                                 ~~~~
    at https://deno.land/x/soxa/src/defaults.ts:59:50

TS7006 [ERROR]: Parameter 'status' implicitly has an 'any' type.
  validateStatus: function validateStatus(status) {
                                          ~~~~~~
    at https://deno.land/x/soxa/src/defaults.ts:80:43

TS7034 [ERROR]: Variable 'parts' implicitly has type 'any[]' in some locations where its type cannot be determined.
    var parts = [];
        ~~~~~
    at https://deno.land/x/soxa/src/helpers/buildURL.ts:33:9

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
    utils.forEach(params, function serialize(val, key) {
                                             ~~~
    at https://deno.land/x/soxa/src/helpers/buildURL.ts:35:46

TS7006 [ERROR]: Parameter 'key' implicitly has an 'any' type.
    utils.forEach(params, function serialize(val, key) {
                                                  ~~~
    at https://deno.land/x/soxa/src/helpers/buildURL.ts:35:51

TS7006 [ERROR]: Parameter 'v' implicitly has an 'any' type.
      utils.forEach(val, function parseValue(v) {
                                             ^
    at https://deno.land/x/soxa/src/helpers/buildURL.ts:46:46

TS7005 [ERROR]: Variable 'parts' implicitly has an 'any[]' type.
    serializedParams = parts.join('&');
                       ~~~~~
    at https://deno.land/x/soxa/src/helpers/buildURL.ts:56:24

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.
      headers[normalizedName] = value;
      ~~~~~~~~~~~~~~~~~~~~~~~
    at https://deno.land/x/soxa/src/helpers/normalizeHeaderName.ts:6:7

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'.
  No index signature with a parameter of type 'string' was found on type '{}'.
      delete headers[name];
             ~~~~~~~~~~~~~
    at https://deno.land/x/soxa/src/helpers/normalizeHeaderName.ts:7:14

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
function isArray(val) {
                 ~~~
    at https://deno.land/x/soxa/src/utils.ts:11:18

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
function isUndefined(val) {
                     ~~~
    at https://deno.land/x/soxa/src/utils.ts:21:22

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
function isBuffer(val) {
                  ~~~
    at https://deno.land/x/soxa/src/utils.ts:31:19

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
function isArrayBuffer(val) {
                       ~~~
    at https://deno.land/x/soxa/src/utils.ts:42:24

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
function isFormData(val) {
                    ~~~
    at https://deno.land/x/soxa/src/utils.ts:52:21

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
function isArrayBufferView(val) {
                           ~~~
    at https://deno.land/x/soxa/src/utils.ts:62:28

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
function isObject(val) {
                  ~~~
    at https://deno.land/x/soxa/src/utils.ts:78:19

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
function isDate(val) {
                ~~~
    at https://deno.land/x/soxa/src/utils.ts:88:17

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
function isFile(val) {
                ~~~
    at https://deno.land/x/soxa/src/utils.ts:98:17

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
function isBlob(val) {
                ~~~
    at https://deno.land/x/soxa/src/utils.ts:108:17

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
function isFunction(val) {
                    ~~~
    at https://deno.land/x/soxa/src/utils.ts:118:21

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
function isStream(val) {
                  ~~~
    at https://deno.land/x/soxa/src/utils.ts:128:19

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
function isURLSearchParams(val) {
                           ~~~
    at https://deno.land/x/soxa/src/utils.ts:138:28

TS7006 [ERROR]: Parameter 'obj' implicitly has an 'any' type.
function forEach(obj, fn) {
                 ~~~
    at https://deno.land/x/soxa/src/utils.ts:154:18

TS7006 [ERROR]: Parameter 'fn' implicitly has an 'any' type.
function forEach(obj, fn) {
                      ~~
    at https://deno.land/x/soxa/src/utils.ts:154:23

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
  function assignValue(val, key) {
                       ~~~
    at https://deno.land/x/soxa/src/utils.ts:200:24

TS7006 [ERROR]: Parameter 'key' implicitly has an 'any' type.
  function assignValue(val, key) {
                            ~~~
    at https://deno.land/x/soxa/src/utils.ts:200:29

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'any' can't be used to index type '{}'.
    if (typeof result[key] === 'object' && typeof val === 'object') {
               ~~~~~~~~~~~
    at https://deno.land/x/soxa/src/utils.ts:201:16

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'any' can't be used to index type '{}'.
      result[key] = merge(result[key], val);
      ~~~~~~~~~~~
    at https://deno.land/x/soxa/src/utils.ts:202:7

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'any' can't be used to index type '{}'.
      result[key] = merge(result[key], val);
                          ~~~~~~~~~~~
    at https://deno.land/x/soxa/src/utils.ts:202:27

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'any' can't be used to index type '{}'.
      result[key] = val;
      ~~~~~~~~~~~
    at https://deno.land/x/soxa/src/utils.ts:204:7

TS7006 [ERROR]: Parameter 'val' implicitly has an 'any' type.
  function assignValue(val, key) {
                       ~~~
    at https://deno.land/x/soxa/src/utils.ts:224:24

TS7006 [ERROR]: Parameter 'key' implicitly has an 'any' type.
  function assignValue(val, key) {
                            ~~~
    at https://deno.land/x/soxa/src/utils.ts:224:29

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'any' can't be used to index type '{}'.
    if (typeof result[key] === 'object' && typeof val === 'object') {
               ~~~~~~~~~~~
    at https://deno.land/x/soxa/src/utils.ts:225:16

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'any' can't be used to index type '{}'.
      result[key] = deepMerge(result[key], val);
      ~~~~~~~~~~~
    at https://deno.land/x/soxa/src/utils.ts:226:7

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'any' can't be used to index type '{}'.
      result[key] = deepMerge(result[key], val);
                              ~~~~~~~~~~~
    at https://deno.land/x/soxa/src/utils.ts:226:31

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'any' can't be used to index type '{}'.
      result[key] = deepMerge({}, val);
      ~~~~~~~~~~~
    at https://deno.land/x/soxa/src/utils.ts:228:7

TS7053 [ERROR]: Element implicitly has an 'any' type because expression of type 'any' can't be used to index type '{}'.
      result[key] = val;
      ~~~~~~~~~~~
    at https://deno.land/x/soxa/src/utils.ts:230:7
```

Some of them can be avoided by having a `tsconfig.json` with `"noImplicitAny: false`, but not all of them.

I am using Deno 1.0.0.

**Comment by jurijtokarski** [comment]
I have the same situation (macOS Mojave).

**Comment by dennisschott** [comment]
Same here.

**Comment by fakoua** [comment]
Issue fixed with Version 1.0.

https://github.com/fakoua/soxa/releases/tag/v1.0

Please add --reload to your cli:

deno run -A --reload YourScript.ts

Thanks.
